### PR TITLE
🎉 New Destination: TiDB

### PR DESCRIPTION
**Pull request by Daemonxiao**
## What
Add new destination [TiDB](https://docs.pingcap.com/tidb/stable) with supporting normalization.

## How
*Describe the solution*
1. Destination: by JDBC connector to transform row data.
2. Normalization: using [dbt-tidb ](https://github.com/pingcap/dbt-tidb)to normalize JSON data.

## Recommended reading order
1. `TiDBDestination.java`
2.  `TiDBSQLNameTransformer.java`
3.  `TiDBSqlOperations.java`
4. `transform.py`
5.  `stream_processor.py`
6. the rest

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [x] Documentation updated
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

</details>

## Tests

<details><summary><strong>Integration</strong></summary>

### Destination Integration Test
N28oapP4CQ

### Normalization Integration Test
```
airbyte$ NORMALIZATION_TEST_TARGET=tidb ./gradlew :airbyte-integrations:bases:base-normalization:integrationTest
```
Pxge8OEhRG

</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Awesome contribution @Daemonxiao I'll ask the team to review next week!

**Comment by Daemonxiao** [comment]
@marcosmarxm It seems something is wrong with pflake8. https://github.com/csachs/pyproject-flake8/issues/13
So I haven't done "pflake" in my work.
How can I fix this error?
```python3 -m pflake8 --config /Users/wq/IdeaProjects/airbyte/pyproject.toml ./
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 188, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 147, in _get_module_details
    return _get_module_details(pkg_main_name, error)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 111, in _get_module_details
    __import__(pkg_name)
  File "/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pflake8/__init__.py", line 54, in <module>
    class ModifiedConfigFileFinder(flake8.options.config.ConfigFileFinder):
AttributeError: module 'flake8.options.config' has no attribute 'ConfigFileFinder'

```

**Comment by Daemonxiao** [comment]
@marcosmarxm Hi, I have fixed the format error. Please trigger the integration test again.

**Comment by Daemonxiao** [comment]
@marcosmarxm I have referred #4910 to fix the integration test. Please trigger the integration test again. Thanks a lot.

File: airbyte-integrations/connectors/destination-tidb/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Destination TiDB
+
+This is the repository for the TiDB destination connector in Java.
+For information about how to use this connector within Airbyte, see [the User Documentation](https://docs.airbyte.io/integrations/destinations/ti-db).
```
Comment (zhangyangyu) [inline airbyte-integrations/connectors/destination-tidb/README.md:4]: `ti-db`?

File: docs/integrations/destinations/tidb.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+# TiDB
+
+[TiDB](https://github.com/pingcap/tidb) (/’taɪdiːbi:/, "Ti" stands for Titanium) is an open-source, distributed, NewSQL database that supports Hybrid Transactional and Analytical Processing (HTAP) workloads. It is TiDB compatible and features horizontal scalability, strong consistency, and high availability. Now, everyone can take a free dev trial on [TiDB Cloud](https://en.pingcap.com/tidb-cloud/).
```
Comment (zhangyangyu) [inline docs/integrations/destinations/tidb.md:3]:
```suggestion
[TiDB](https://github.com/pingcap/tidb) (/’taɪdiːbi:/, "Ti" stands for Titanium) is an open-source, distributed, NewSQL database that supports Hybrid Transactional and Analytical Processing (HTAP) workloads. It is MySQL compatible and features horizontal scalability, strong consistency, and high availability. Now, everyone can take a free dev trial on [TiDB Cloud](https://en.pingcap.com/tidb-cloud/).
```

File: airbyte-integrations/connectors/destination-tidb/README.md (hunk 1 above)
Comment (Daemonxiao) [reply airbyte-integrations/connectors/destination-tidb/README.md:4]: @zhangyangyu Thanks for your review. I will correct it.

**Review by zhangyangyu — APPROVED** [review]

**Comment by Daemonxiao** [comment]
@marcosmarxm Integration test reported that `test_normalization[DestinationType.MSSQL-test_simple_streams]` was failed. Does something wrong with the MSSQL-normalization docker image? Plz, take a look.

**Comment by marcosmarxm** [comment]
@Daemonxiao yep, I'll take a look. Looks your contributation is close to be merged!

**Review by marcosmarxm — APPROVED** [review]
thanks @Daemonxiao for this amazing contribution

**Pull request merged**